### PR TITLE
Page kit migration

### LINKED ISSUE #123: Migrate next-b2b-prospect to Page Kit 🎉
> 🤖 This issue was automatically created as part of the [Page Kit tracking project](https://github.com/orgs/Financial-Times/projects/75).

The `@financial-times/n-ui` tooling is going to be deprecated, in favour of the [Financial-Times/dotcom-page-kit](https://github.com/Financial-Times/dotcom-page-kit) project.

Page Kit...

> Provides a high quality, well tested, and thoroughly documented set of tools for assembling and delivering FT.com based upon the best industry standards.

Which is a significant improvement on the [Financial-Times/n-ui tooling](https://github.com/Financial-Times/n-ui).

This is repository should follow [the Page Kit migration guide](https://github.com/Financial-Times/dotcom-page-kit/blob/master/docs/guides/migrating-from-n-ui-to-page-kit.md) to remove the dependency.

### PR DESCRIPTION
**Pull request by adambraimbridge**
Fixes #123 

This PR replaces n-ui with Page Kit 0.4.1.  No user-facing functionality should have changed for this app.

### CODE REVIEW

**Comment by jenniferemshepherd** [comment]
This is ready bar a couple of fixes that are being merged into the page kit branch [here](https://github.com/Financial-Times/dotcom-page-kit/pull/656).  That code is more advanced than the currently released version of page kit however, so we need to be careful pulling any of it in here.

File: package.json [hunk 1]
```diff
@@ -16,13 +16,18 @@
     "prepublish": "npm run snyk-protect"
   },
   "dependencies": {
+    "@financial-times/dotcom-middleware-assets": "^0.3.2",
+    "@financial-times/dotcom-middleware-navigation": "^0.3.2",
+    "@financial-times/dotcom-server-handlebars": "^0.2.3",
```
Comment (i-like-robots) [inline package.json]: Please make sure all package versions are the same 🔎

File: server/page-kit.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import ReactDOM from 'react-dom/server';
+import { Shell } from '@financial-times/dotcom-ui-shell';
+import { Layout } from '@financial-times/dotcom-ui-layout';
+
+export default ({ response, next, shellProps, layoutProps }) => {
+        layoutProps.navigationData = response.locals.navigation;
+        layoutProps.headerOptions = { ...response.locals.anon};
```
Comment (i-like-robots) [inline server/page-kit.js]: I think the scripts and stylesheets config could be abstracted here as well as it looks to be the same everywhere

File: server/server.js [hunk 3]
```diff
@@ -8,20 +9,38 @@ import path from 'path';
...
 const PORT = Number(process.env.PORT || 5657);
 
 const app = express({
 	systemCode: 'next-b2b-prospect',
+	appName: 'b2b-prospect',
 	graphiteName: 'b2b-prospect',
+	withConsent: true,
+	withAnonMiddleware: false,
+	withFlags: false,
 	withJsonLd: false,
```
Comment (i-like-robots) [inline server/server.js]: The `withJsonLd` option can be removed as this is an `n-ui` option.

File: server/modules/form/api.js [hunk 4]
```diff
@@ -46,7 +62,7 @@ export default {
 
 		const emailValue = await getUserEmail(req.cookies.FTSession_s);
 
-		return res.render('form', {
+		const templateData = {
 			title: 'Signup',
```
Comment (pixelandpage) [inline server/modules/form/api.js]: There are a few places in this file where we're hardcoding 'Signup' as a title property. We could read the title from this `templateData` object instead in those places 🙂

File: Makefile [hunk 5]
```diff
@@ -25,7 +22,7 @@ unit-test:
 	export MEMBQL_API_KEY_PROD=1; \
 	mocha --require server/setup --exit --require test/setup --recursive ./test
 
-run:
+run-local:
```
Comment (pixelandpage) [inline Makefile]: Should this stay as `run`?

File: package.json [hunk 6]
```diff
@@ -31,15 +36,26 @@
     "n-common-static-data": "github:Financial-Times/n-common-static-data.git#v1.6.2",
     "n-health": "3.3.1",
     "node-marketo-rest": "^0.7.0",
-    "snyk": "^1.163.3"
+    "react": "^16.11.0",
+    "react-dom": "^16.11.0",
+    "snyk": "^1.163.3",
+    "sucrase": "^3.10.1"
```
Comment (pixelandpage) [inline package.json]: We can make `sucrase` a devDependency to keep it with the other testing packages

Comment (jenniferemshepherd) [reply package.json]: no we can't because we are using it in production ;)

**Comment by jenniferemshepherd** [comment]
💥

**Pull request merged**